egg/platform_x86.py

```diff
@@ -738,22 +883,13 @@ def binlop2(func, simd_ext, typ):
                       ret.v1 = (__mmask16)({in0}.v1 {op} {in1}.v1);
                       return ret;'''. \
                       format(op=op[func], op_fct=op_fct[func], **fmtspec)
+        elif simd_ext == 'avx512_knl' and fmtspec['le'] != 16:
+            # KNL only has the 'mask16' variant
```

Comment:
I do not understand the comment that seems to be the "not" of the condition of the `if` above.

Reply:
KNL only has the 16-bit variant, hence we cannot call `_{op_fct}_mask{le}` and must use a fallback instead. The fallback uses `int` operations (which might have to move the values to scalar registers instead of keeping them in the mask registers).

I will update the comment.